### PR TITLE
UPDATED Feature 7 - playSong() - needs to be tested, minor changes and additions to some of the skeletal files

### PR DESCRIPTION
**Pull request by Mistryr222**
1. Done use cases for playSong() and shufflePlaylist()
2. Changes to input and output data skeletal files
3. Small technical details changed in files like the Song entity
4. Commented out @sarxhlwalker's code for her to make a change on the use case

### CODE REVIEW

File: src/main/java/Controller/playAudioController.java [hunk 1]
```diff
@@ -24,9 +24,9 @@ public playAudioController(playAudioInputBoundary playAudioInteractor) {
      * Function calling the use case for playing audio
      * @param songFile is the File object of the MP3 file
      */
-    public void playAudio(File songFile) {
-        audioInputData audioInput = new audioInputData(songFile);
-        this.playAudioInteractor.playAudio(audioInput);
+    public void playAudio(songInputData songFile) {
+        songInputData songInput = new songInputData(songFile.getSong());
```
Comment (sarxhlwalker) [inline src/main/java/Controller/playAudioController.java:28]: Why are you creating a new songInputData when the parameter passed to playAudio is a songInputData already?

Comment (sarxhlwalker) [reply src/main/java/Controller/playAudioController.java:28]: Also, what's the difference between playAudioController and Raf's musicEngineController?

File: src/main/java/Controller/playSongController.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * Controller for the playSong use case
+ */
+public class playSongController {
+
+    private final playSongInputBoundary playSongInteractor;
+    private final songInputData songInputData;
+
+    public playSongController(playSongInputBoundary playSongInteractor, Song song) {
+        this.playSongInteractor = playSongInteractor;
```
Comment (sarxhlwalker) [inline src/main/java/Controller/playSongController.java:16]: After the conversation we had in tutorial today, I was under the impression that the controller will be _creating_ an instance of the interactor, as opposed to being passed one.

File: src/main/java/Controller/playlistController.java [hunk 3]
```diff
@@ -1,4 +1,26 @@
...
+
+/**
+ * Controller for the shufflePlaylist use case
+ */
 public class playlistController {
+
+    private final shufflePlaylistInputBoundary shufflePlaylistInteractor;
+
+    public playlistController(shufflePlaylistInputBoundary shufflePlaylistInteractor) {
+        this.shufflePlaylistInteractor = shufflePlaylistInteractor;
```
Comment (sarxhlwalker) [inline src/main/java/Controller/playlistController.java:18]: same as prev. comment.

File: src/main/java/Entities/Song.java [hunk 4]
```diff
@@ -26,7 +26,7 @@ public class Song {
      * @param isExplicit Whether the song is explicit or not.
      * @param cover Cover image of the song.
      */
-    public Song(int id, String name, List artistList, int length, String genre,
+    public Song(int id, String name, List<ArtistUser> artistList, int length, String genre,
```
Comment (sarxhlwalker) [inline src/main/java/Entities/Song.java:29]: Thanks for making these changes!

File: src/main/java/InputBoundary/playAudioInputBoundary.java [hunk 5]
```diff
@@ -1,15 +1,12 @@
...
-import java.io.FileNotFoundException;
+import InputData.songInputData;
 
 /**
  * Interface for the playAudioInteractor to implement for the use case
  */
 public interface playAudioInputBoundary {
 
-    void playAudio(audioInputData songFile);
+    void playAudio(songInputData song);
```
Comment (sarxhlwalker) [inline src/main/java/InputBoundary/playAudioInputBoundary.java:10]: Just to clarify, is this the method that I will be calling? (playAudio vs. playSong)

File: src/main/java/InputData/songInputData.java [hunk 6]
```diff
@@ -1,7 +1,58 @@
...
+
+    public songInputData(Song song){
+        this.song = song;
+    }
+
+    public Song getSong() {
+        return song;
+    }
+
+    public int getId() {
```
Comment (sarxhlwalker) [inline src/main/java/InputData/songInputData.java:21]: Thanks for implementing these functions! I think it's a great idea as opposed to calling the Song entity's .getID function itself.

File: src/main/java/UseCase/playSongInteractor.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    private final MusicPlayer mp = MusicPlayer.getInstance();
+    private final songOutputBoundary songPresenter;
+    private final Song song;
+
+    public playSongInteractor(songInputData s, songOutputBoundary songPresenter){
+        this.song = s.getSong();
+        this.songPresenter = songPresenter;
+    }
+
+    public void playAudio() {
```
Comment (sarxhlwalker) [inline src/main/java/UseCase/playSongInteractor.java:22]: This is where I'm confused -- there is both the playAudio via the playAudioInteractor, and the playAudio via the playSongInteractor. Which should I be interacting with? I assume playAudioInteractor, in which case this function could be deleted. Please let me know if this is incorrect.

File: src/main/java/UseCase/playSpaceInteractor.java [hunk 8]
```diff
@@ -30,7 +30,8 @@ private void playSpace(spaceInputData spaceInputData){
         File songFile = song.getFile();
 
         // play song
-        playSong.playAudio(songFile);
+        // playAudioInteractor audio = new playAudioInteractor();
```
Comment (sarxhlwalker) [inline src/main/java/UseCase/playSpaceInteractor.java:33]: Thanks for commenting this out for the autograde! I'll be sure to fix this in my next PR.

**Review by sarxhlwalker — COMMENTED** [review]
Great work! I have a few clarifying comments and such before I approve.

**Comment by rafaelgacesa** [comment]
### _**Looks good!!!**_

**Review by rafaelgacesa — APPROVED** [review]
### _**Looks good!!!**_

**Review by chrstinalin — APPROVED** [review]
### Looks great!!

**Pull request merged**